emmet-builders/emmet/builders/vasp/thermo.py

```diff
@@ -99,57 +100,54 @@ def get_items(self) -> Iterator[List[Dict]]:
         # Will build them in a similar manner to fast Pourbaix
         for chemsys in sorted(to_process_chemsys, key=lambda x: len(x.split("-"))):
             entries = self.get_entries(chemsys)
-
-            # build sandbox sets: ["a"] , ["a","b"], ["core","a","b"]
-            sandbox_sets = set(
-                [frozenset(entry.data.get("sandboxes", {})) for entry in entries]
-            )
-            sandbox_sets = maximal_spanning_non_intersecting_subsets(sandbox_sets)
-            self.logger.debug(f"Found {len(sandbox_sets)}: {sandbox_sets}")
-
-            for sandboxes in sandbox_sets:
-                # only yield maximal subsets so that we can process a equivalent sandbox combinations at a time
-                sandbox_entries = [
-                    entry
-                    for entry in entries
-                    if all(
-                        sandbox in entry.data.get("_sbxn", []) for sandbox in sandboxes
-                    )
-                ]
-
-                yield sandboxes, sandbox_entries
+            yield entries
+            # # build sandbox sets: ["a"] , ["a","b"], ["core","a","b"]
+            # sandbox_sets = set(
+            #     [frozenset(entry.data.get("sandboxes", {})) for entry in entries]
+            # )
+            # sandbox_sets = maximal_spanning_non_intersecting_subsets(sandbox_sets)
+            # self.logger.debug(f"Found {len(sandbox_sets)}: {sandbox_sets}")
+            #
+            # for sandboxes in sandbox_sets:
+            #     # only yield maximal subsets so that we can process a equivalent sandbox combinations at a time
+            #     sandbox_entries = [
+            #         entry
+            #         for entry in entries
+            #         if all(
+            #             sandbox in entry.data.get("_sbxn", []) for sandbox in sandboxes
+            #         )
+            #     ]
+            #
+            #     yield sandboxes, sandbox_entries
 
     def process_item(self, item: Tuple[List[str], List[ComputedEntry]]):
 
-        sandboxes, entries = item
+        entries = item
 
-        entries = [ComputedEntry.from_dict(entry) for entry in entries]
+        entries = [ComputedStructureEntry.from_dict(entry) for entry in entries]
         # determine chemsys
         elements = sorted(
             set([el.symbol for e in entries for el in e.composition.elements])
         )
         chemsys = "-".join(elements)
 
-        self.logger.debug(
-            f"Procesing {len(entries)} entries for {chemsys} - {sandboxes}"
-        )
+        self.logger.debug(f"Procesing {len(entries)} entries for {chemsys}")
 
-        material_entries = defaultdict(defaultdict(list))
+        material_entries = defaultdict(lambda: defaultdict(list))
```

Comment:
Can you clarify why the lambda is needed?

Reply:
I think that's just how `defaultdict` works. 
You need a callable function which returns something and `lambda : x` is just a function that returns x with no inupt.
So if a key is missing it needs to call that function to get the default.
So `defaultdict(lambda : 2)` is how you would get a default value of 2.